web-components/button/button.svelte

```diff
@@ -105,7 +116,7 @@
     --color-loading: var(--color-gray-70);
     --border-width: 1px;
     --border-color: var(--color-gray-30);
-    &[disabled] {
+    &[disabled=true] {
```

Comment:
Actually will this work for `<button disabled>Foo</button>` (because the button doesn't have the attribute `disabled=true`)?

Reply:
`&[disabled]` would only work to determine the presence of the `disabled` attribute, and since simply setting `disabled` on an element (e.g. `<a disabled ...>`) is sufficient to make it truthy, that could work.

However in our case, the `disabled` attribute is always present, and set either to true or false: https://github.com/brave-experiments/leo/blob/main/web-components/button/button.svelte#L140

Reply:
We could arguably change this selector to be `&:disabled`.

Reply:
So I think this actually breaks because of the way boolean attributes are compiled to HTML.

```svelte
<button disabled={true}>Foo</button>
// compiles to
<button disabled>Foo</button>

// while
<button disabled={false}>Foo</button>
// compiles to
<button>Foo</button>
```

I think, any value for the attribute is considered "truthy", which kind of makes sense from a JS perspective - (!!"true" is true, !!"false" is true).

So for example, the following button would actually be disabled:
```html
<button disabled="false">Foo</button>
```
even though disabled is `"false"`.

If you run the button Storybook at the moment, the disabled toggle actually isn't working.

HTML is weird as :laughing:

Reply:
Yeah, so I think we need to change the prop from `isDisabled` to `disabled` and then set it on the component like this:

```svelte
<svelte:element
  this={tag}
  {href}
  ...
  {disabled}
  ...
>
  <slot>Leo Button</slot>
</svelte:element>

Reply:
Button, not disabled:
![Screenshot from 2022-09-20 10-22-29](https://user-images.githubusercontent.com/7678024/191130018-756ee45e-da1b-48f2-a2bb-8641f83b8f19.png)

Button, disabled
![Screenshot from 2022-09-20 10-23-55](https://user-images.githubusercontent.com/7678024/191130150-f50956a9-9864-4ee9-b6d2-9a5fd6ad3903.png)

Reply:
Okay, I see what you were trying to solve now :laughing: Sorry about that. 

So basically, it looks like `<a>` tags don't support being disabled, which is what's causing our problems (Svelte isn't handling the disabled prop properly). This actually means we have a bunch more problems, because when you set `disabled=true` the link is still clickable (maybe we need to add a `pointer-events: none` to the link when disabled, or something cleverer).

For the styling, I think we might need `[disabled=true], [disabled] to get the selector to work for disabled buttons and disabled links.

Reply:
I think if we change our component to not always set `disabled` to true or false, we'll be good to go, and can style using `:disabled`.

https://svelte.dev/repl/cb5d5167a0e441bda676227ca7d2f4e5?version=3.50.1

Reply:
I'm pretty sure that will break `disabled=true` for the `a` button

Reply:
I don't follow? We only need to support `disabled=true` for `button`, right?

Reply:
Ah, okay, so we're fine not having a disabled mode for links?

Reply:
Totally, since that would be non-standard HTML.

Reply:
Sweet, sounds good to me!